src/cortexpy/tesserae.py

```diff
@@ -78,9 +78,9 @@ def __initialize(self, query, targets):
         self.maxl = max_length(query, targets)
         self.qlen = len(query)
 
-        self.vt_m = np.full([self.nseq, self.maxl + 1, 2], SMALL, dtype=np.float64)
-        self.vt_i = np.full([self.nseq, self.maxl + 1, 2], SMALL, dtype=np.float64)
-        self.vt_d = np.full([self.nseq, self.maxl + 1, 2], SMALL, dtype=np.float64)
+        self.vt_m = np.full([2, self.nseq, self.maxl + 1], SMALL, dtype=np.float64)
+        self.vt_i = np.full([2, self.nseq, self.maxl + 1], SMALL, dtype=np.float64)
+        self.vt_d = np.full([2, self.nseq, self.maxl + 1], SMALL, dtype=np.float64)
```

Comment:
Can we please have descriptive named variables for the matrix dimensions. That would help understanding the code more quickly, and make these sort of changes a bit easier.

Reply:
It certainly would, wouldn't it? I think the reason I did it like this (and with the `l0` and `l1` later commented) was to keep up with the "style that already was there". But that's of course no excuse for unclean code...

B.t.w., there are some further memory optimizations possible in the matrix sizes, for instance, the variable `self.maxl` takes the query length into consideration, although the code acually only deals with the targets in the matrix. Was the query inteded to be used this way somehow, but it wasn't implemented for some reason?

Reply:
I completely agree that sticking with the style of the code that was already there is a great rule. However, as I think you would agree, every time we make changes to code should be a time when we try to leave it in a cleaner state after our changes. I think the tesserae code was always in need of some cleaning up. Perhaps it even needs a rewrite. I want this code to be maintainable if we are to keep it in the cortexpy library going forward. What do you think is the best way to achieve that @karljohanw? What small changes could you make now/here that would make a rewrite/major refactoring easier to do for the next coder who might not be familiar with this code?

Reply:
2/2 regarding the other optimizations, I think once we have cleaner code these sort of optimizations will become more obvious, and of course we should pursue therm.

Reply:
I can't speak to particular design choices for this code. @kvg?